File: fontend/src/components/Services/FieldService.js
import axios from 'axios'

// Author: Phạm Văn Đạt
// function: Service xử lý thao tác lĩnh vực
//  created time: 11:00 16/08/2022
export class FieldService{

    url= "https://localhost:44355/api/Field";
    field = "";

    constructor(){
        
    }

    // lấy danh sách Lĩnh vực
    async GetAll(){
       try{
         return axios.get(this.url).then((res) => {
                 return res.data.data;
              })
              .catch((e) => {
                console.log(e);
              });     

       }catch(error){
        console.log(error);
       }
    }

    // lấy danh sách lĩnh vực theo tên
    async GetByName(search){
        try{
            return axios.get(this.url+search).then((res) => {
                return res.data.data;
             })
             .catch((e) => {
               console.log(e);
             });         
        }catch(error){
            console.log(error);
        }
    }
}

File: backend/WebApplication/WebDomain/Services/FieldService/FieldService.cs
using Common.Resources;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WebInfrastructure;

namespace WebDomain
{
    /// <summary>
    /// Author: Phạm Văn Đạt
    /// Created Time: 4:23 15/08/2022
    /// function: Service xử lý bảng lĩnh vực
    /// </summary>
    public class FieldService : IFieldService
    {

        private readonly IDapperRepository _dapper;
        public FieldService(IDapperRepository dapper)
        {
            _dapper = dapper;
        }
        public async Task<ReponsitoryModel> GetAll()
        {
            try
            {
                string sql = @"SELECT FieldId,FieldName FROM Field";
                List<FieldModel> result = await _dapper.GetAllAsync<FieldModel>(sql);
                if (result == null)
                    return new ReponsitoryModel { Data = null, Message = MessageError.NotValue, StatusCode = CodeError.Code400 };
                return new ReponsitoryModel { Data = result, Message = MessageSuccess.GetSuccess, StatusCode = CodeSuccess.Status200 };
            }
            catch (Exception ex)
            {
                return new ReponsitoryModel { Data = ex.Message, Message = MessageError.ProcessError, StatusCode = CodeError.Code500 };
            }
        }

        public async Task<ReponsitoryModel> GetAllByName(string name)
        {
            try
            {
                string sql = @"SELECT FieldId,FieldName FROM Field WHERE FieldName like @search;";
                var parameters = new Dictionary<string, object>()
                {
                    ["search"] = $"%{name}%"
                };
                List<FieldModel> result = await _dapper.FindTAsync<FieldModel>(sql, parameters);
                if (result.Count == 0)
                    return new ReponsitoryModel { Data = null, Message = MessageError.NotValue, StatusCode = CodeError.Code400 };
                return new ReponsitoryModel { Data = result, Message = MessageSuccess.GetSuccess, StatusCode = CodeSuccess.Status200 };
            }
            catch (Exception ex)
            {
                return new ReponsitoryModel { Data = ex, Message = MessageError.ProcessError, StatusCode = CodeError.Code500 };
            }
        }
    }
}
